netutils/interface.py

```diff
@@ -172,3 +175,148 @@ def abbreviated_interface_name(interface, addl_name_map=None, addl_reverse_map=N
         raise ValueError(f"Verify interface on and no match found for {interface}")
     # If abbreviated name lookup fails, return original name
     return interface
+
+
+def interface_range_compress(
+    interfaces: Union[str, List[str]] = None, prefix: str = "interface range ", max_ranges: int = 5
+) -> List[str]:  # noqa: R0914, R0912, R0915
+    """Function which takes interfaces and return interface ranges.
+
+    By default this function creates Cisco `interface range ...` commands.
+    Whitespace and special characters are ignored in the input. Input must contain only interfaces,
+    there is no check against correct interface names!
+
+    Example:
+        >>> interface_range_compress("Gi1/0/1 Gi1/0/2 Gi1/0/4")
+        ['interface range Gi1/0/1-2, Gi1/0/4']
+        >>> interface_range_compress(["Gi1/0/1 Gi1/0/2","Gi1/0/3,Gi1/0/4"])
+        ['interface range Gi1/0/1-4']
+
+    Args:
+        prefix: prefix string used in each line of the output
+        interfaces: interfaces in a line or in a list of lines
+        max_ranges: maximum number of ranges in one line in the output
+
+    Returns:
+        list: list of interface ranges
+    """
+    Port = namedtuple("Port", ["iface_name", "module1", "module2", "module3", "module4"])
+    """ Port type which contains the interface name and submodules/ports.
+
+    iface_name: str - name of interface (Gi, Fa, etc..)
+    module1..4: int - number of submoduled or port. -1 indicates that it is not part of the
+                      interface
+                      4 module depth is supported which should cover most cases.
+    """  # noqa: W0105
+
+    def assemble_port(port_in: Port):
+        """Assemble exploded port_in.
+
+        Separator is `/`. -1 value means that value is not used.
+        Only physical ports are supported.
+
+        Example:
+            Gi, 1, 0, 1, -1 will become: Gi1/0/1
+
+        Args:
+            port_in: tuple of port name and fex, chasses, module, interface numbers
+
+        Returns:
+            string of assembled interface
+        """
+        out = port_in.iface_name
+        if port_in.module1 >= 0:
+            out += str(port_in.module1)
+            if port_in.module2 >= 0:
+                out += "/" + str(port_in.module2)
+                if port_in.module3 >= 0:
+                    out += "/" + str(port_in.module3)
+                    if port_in.module4 >= 0:
+                        out += "/" + str(port_in.module4)
+        return out
+
+    ports = []  # collect exploded interfaces
+    # case insensitive port parsing. We do a hard assumption that we only have ports in the input.
+
+    output = []
+    if interfaces is None:
+        return []
+    # read all lines and explode all interfaces as preparation for sorting
+    for line in [interfaces] if isinstance(interfaces, str) else interfaces:
+        # matches = port_regex.findall(line)
```

Comment:
Can this be removed? Might have been part of the development process.

Reply:
Cleaned up in [last commit](https://github.com/networktocode/netutils/pull/33/commits/8f9a51aaa0357c92a89eb88a5ae84c2797c2c359#diff-e880d02b86528af00499f4d00b3d768decbd9ff9d6c5921ee90479c468e5e043L246)